src/main/java/com/jnape/palatable/lambda/monoid/builtin/MapMerge.java

```diff
@@ -0,0 +1,47 @@
+package com.jnape.palatable.lambda.monoid.builtin;
+
+import com.jnape.palatable.lambda.functions.Fn1;
+import com.jnape.palatable.lambda.functions.specialized.BiMonoidFactory;
+import com.jnape.palatable.lambda.functions.specialized.MonoidFactory;
+import com.jnape.palatable.lambda.monoid.Monoid;
+import com.jnape.palatable.lambda.semigroup.Semigroup;
+
+import java.util.Map;
+import java.util.function.Supplier;
+
+import static com.jnape.palatable.lambda.monoid.Monoid.monoid;
+
+public class MapMerge<A, B> implements BiMonoidFactory<Supplier<Map<A, B>>, Semigroup<B>, Map<A, B>> {
+    private MapMerge() {
+    }
+
+    @Override
+    public Monoid<Map<A, B>> apply(Supplier<Map<A, B>> mSupplier, Semigroup<B> semigroup) {
+        return monoid((x, y) -> {
+            Map<A, B> copy = mSupplier.get();
+            copy.putAll(x);
+            y.forEach((a, b) -> copy.merge(a, b, semigroup.toBiFunction()));
+            return copy;
+        }, mSupplier.get());
```

Comment:
Rather than explicitly invoking `.get()` here, let's defer that and use the `monoid` overload where the second argument is a `Supplier`. This will likely break inference, so you may need to ascribe the call on line 20, ala, `Monoid.<Map<A,B>>monoid`, but at least then we won't be forcing the `Supplier` until necessary.